llvm_objects.py

```diff
@@ -196,46 +242,36 @@ def _get_operands(self, l_instr, w_module):
         args = []
         for i in range(llwrap.LLVMGetNumOperands(l_instr)):
             arg = llwrap.LLVMGetOperand(l_instr, i)
-            args.append(arg)
-            if llwrap.LLVMIsConstant(arg):
-                addr = rffi.cast(rffi.INT, arg)
-                var_type = llwrap.LLVMGetTypeKind(llwrap.LLVMTypeOf(arg))
-                if var_type == llwrap.LLVMIntegerTypeKind:
-                    if llwrap.LLVMIsUndef(arg):
-                        # XXX undefined vars -- currently set to 0
-                        w_module.constants.set_variable(addr, Integer(0))
-                    else:
-                        w_module.constants.set_variable(addr, Integer(llwrap.LLVMConstIntGetSExtValue(arg)))
-                elif var_type == llwrap.LLVMDoubleTypeKind or var_type == llwrap.LLVMFloatTypeKind:
-                    if llwrap.LLVMIsUndef(arg):
-                        w_module.constants.set_variable(addr, Float(0.0))
-                    else:
-                        with lltype.scoped_alloc(rffi.SIGNEDP.TO, 1) as signed_ptr:
-                            fl_var = Float(llwrap.LLVMConstRealGetDouble(arg, signed_ptr))
-                            w_module.constants.set_variable(addr, fl_var)
+            args.append(_get_variable_wrapper(arg))
         return args
 
 class W_BrInstruction(W_BaseInstruction):
     ''' Represents a wrapper for an LLVMValueRef that is a br instruction. '''
 
+    _immutable_fields_ = ['w_bb_uncond']
+
     def __init__(self, l_instr, w_module):
         W_BaseInstruction.__init__(self, l_instr, w_module)
-        self.l_bb_uncond = llwrap.LLVMValueAsBasicBlock(self.l_operands[0])
+        self.w_bb_uncond = None
```

Comment:
I don't know if it's safe in this case. If you have verified that it is *definitely* safe, and also know that it will always be safe in the future, keep it. Otherwise, unless you're 100% sure, remove the immutable field, as it's potentially very dangerous.